src/aggregators/unwrap-tuple-providers.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { PropsWithChildren, ReactElement } from 'react';
+import { ProviderComponent, TupleProviderWithProps } from '../types/provider';
+
+export function unWrapTupleProvider(
+  tuple: ProviderComponent<unknown> | TupleProviderWithProps,
+): ProviderComponent<unknown> {
+  if (!Array.isArray(tuple)) {
+    return tuple;
+  }
+
+  const [Component, props] = tuple;
+
+  return function CurriedComponent({
```

Comment:
@davelosert do you know a way for this component to take the name of the unwrapped component?

Reply:
@sincovschi : I don't quite follow - what do you mean by taking the name? Maybe you can elaborate with a code example?

Reply:
If you look to components tree into React browser's extention. It will look something like this:

```
<CurriedComponent>
  <CurriedComponent>
    <CurriedComponent>
      <Component1>
        <Component2>
.......
```

What I want that `<CurriedComponent>` to be treated as Original component's name.

Reply:
Hmmm... not sure. I actually dont know where the DEV-Tools take the information from.
Maybe we can add a certain property to the function? Or somehow name the function dynamially?

I will have to try some things out, can not say if this is possible.

Reply:
Hmmm didn't find a quick solution. There is this property `displayName` for a Component - but I'm not sure right now if this could be the solution as I do not have a test-setup.

So my proposal would be to add this to the improvements-list so that we can get this branch merged?

Reply:
Yeah, that I also found and tried without success. So let's keep it as it is. Thanks.